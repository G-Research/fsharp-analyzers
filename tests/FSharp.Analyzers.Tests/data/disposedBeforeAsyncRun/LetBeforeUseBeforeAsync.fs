module M

open System

type DisposableThing () =
    interface IDisposable with
        member _.Dispose() = ()

let asyncReturningFunc () =
    let x = 2 + 3
    // should warn
    use theThing = new DisposableThing()
    async {
        return "hi"
    }
