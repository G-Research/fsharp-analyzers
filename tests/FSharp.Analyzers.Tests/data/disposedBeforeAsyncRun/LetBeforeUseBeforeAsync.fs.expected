GRA-DISPBEFOREASYNC-001 | Warning | (12,8 - 12,16) | Object is disposed before returned async is run | []
